test/controllers/orderitems_controller_test.rb

```diff
@@ -0,0 +1,178 @@
+require "test_helper"
+
+describe OrderitemsController do
+
+  describe 'create' do
+    it 'can add a valid orderitem' do
+      oi_data = {
```

Comment:
I would include separate tests here for if the cart already exists or if it should create a new cart